apps/examples/smartfs/read_optimization/read_optimization_main.c

```diff
@@ -0,0 +1,273 @@
+		fprintf(stderr, "ERROR: Failed to open Free Run Timer: %d\n", errno);
+		return -1;
+	}
+
+	if (ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN) < 0) {
+		fprintf(stderr, "ERROR: Failed to set Free Run Timer: %d\n", errno);
+		return -1;
+	}
+	if (ioctl(frt_fd, TCIOC_START, TRUE) < 0) {
+		fprintf(stderr, "ERROR: Failed to start Free Run Timer: %d\n", errno);
+		return -1;
+	}
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(512);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Throughput while reading %dKb file with 512 byte buffer = %.02fKbps\n", size, (float)(size * 1000000) / time_read);
+	printf("\n****************************************************************************************\n");
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(1024);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Throughput while reading %dKb file with 1024 byte buffer = %.02fKbps\n", size, (float)(size * 1000000) / time_read);
+	printf("\n****************************************************************************************\n");
+
+	ret = stat("/mnt", &st);
+	if (ret < 0) {
+		printf("Stat to get optimal read buffer size failed, Errno = %d\n", errno);
+		return -1;
+	}
+	buf_size = st.st_blksize;
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(buf_size);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Throughput while reading %dKb file with %d byte buffer = %.02fKbps\n", size, buf_size, (float)(size * 1000000) / time_read);
+	printf("\n****************************************************************************************\n");
+
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Free Run Timer: %d\n", errno);
+	}
+
```

Comment:
I think we need to add close(frt_fd) before exiting app.

Reply:
Done, thankyou